catalog/app/containers/Bucket/PackageUpdateDialog.js

```diff
@@ -0,0 +1,933 @@
+          This package is not editable via the web UI&mdash;it cannot handle package
+          manifest that large ({readableBytes(e.actualSize)}).
+        </M.Typography>
+        <M.Typography>Please use Quilt CLI to edit this package.</M.Typography>
+      </>
+    ),
+  ],
+  [
+    R.T,
+    () => (
+      <>
+        <M.Typography variant="h6" gutterBottom>
+          Unexpeceted error
+        </M.Typography>
+        <M.Typography gutterBottom>
+          Something went wrong. Please contact Quilt support.
+        </M.Typography>
+        <M.Typography>You can also use Quilt CLI to edit this package.</M.Typography>
+      </>
+    ),
+  ],
+])
+
+const useDialogErrorStyles = M.makeStyles((t) => ({
+  overlay: {
+    background: fade(t.palette.common.white, 0.4),
+    bottom: 0,
+    display: 'flex',
+    flexDirection: 'column',
+    left: 0,
+    padding: t.spacing(2, 3, 4),
+    position: 'absolute',
+    right: 0,
+    top: 0,
+  },
+}))
+
+function DialogError({ error, close }) {
+  const classes = useDialogErrorStyles()
+  return (
+    <>
+      <M.DialogTitle>Push package revision</M.DialogTitle>
+      <M.DialogContent style={{ paddingTop: 0, position: 'relative' }}>
+        <PD.FormSkeleton animate={false} />
+        <div className={classes.overlay}>{errorDisplay(error)}</div>
+      </M.DialogContent>
+      <M.DialogActions>
+        <M.Button onClick={close}>Cancel</M.Button>
+        <M.Button variant="contained" color="primary" disabled>
+          Push
+        </M.Button>
+      </M.DialogActions>
+    </>
+  )
+}
+
+function DialogSuccess({ bucket, name, revision, close }) {
+  const { urls } = NamedRoutes.use()
+  return (
+    <>
+      <M.DialogTitle>Push complete</M.DialogTitle>
+      <M.DialogContent style={{ paddingTop: 0 }}>
+        <M.Typography>
+          Package revision{' '}
+          <StyledLink to={urls.bucketPackageTree(bucket, name, revision)}>
+            {name}@{revision}
+          </StyledLink>{' '}
+          successfully created
+        </M.Typography>
+      </M.DialogContent>
+      <M.DialogActions>
+        <M.Button onClick={close}>Close</M.Button>
+        <M.Button
+          component={Link}
+          to={urls.bucketPackageTree(bucket, name, revision)}
+          variant="contained"
+          color="primary"
+        >
+          Browse
+        </M.Button>
+      </M.DialogActions>
+    </>
+  )
+}
+
+const DialogState = tagged([
+  'Closed',
+  'Loading',
+  'Error',
+  'Form', // { manifest, workflowsConfig }
+  'Success', // { name, revision }
+])
+
+export function usePackageUpdateDialog({ bucket, name, revision, onExited }) {
+  const s3 = AWS.S3.use()
+
+  const [isOpen, setOpen] = React.useState(false)
+  const [wasOpened, setWasOpened] = React.useState(false)
+  const [exited, setExited] = React.useState(!isOpen)
+  const [exitValue, setExitValue] = React.useState(null)
+  const [success, setSuccess] = React.useState(false)
+  const [key, setKey] = React.useState(1)
```

Comment:
It's not a AWS Key, it's not a React key, it's not an object key, right? It's something from `utils`, and it looks like not implemented yet?
Maybe use a more specific name?

Reply:
it's not from utils and it is implemented. this is used as a sort of cache-busting, a way to trigger component re-renders and data re-fetching.

Reply:
As I see from code, `key` fed to `requests.loadManifest` through `useData`. But `loadManifest` don't expect `key` argument. That's why I said, that it looks like not implemented.

Reply:
`key` is used by `useMemoEq`, right?

Reply:
> As I see from code, key fed to requests.loadManifest through useData. But loadManifest don't expect key argument. That's why I said, that it looks like not implemented.

it's not used by the requests function, but it is used by the `useData` hook itself which remembers the parameters dict and re-fetches when it's changed

> And also because there is TODO note about key in useData

there's no custom key extraction feature, so the whole parameters dict is used by default

> key is used by useMemoEq, right?

yep: https://github.com/quiltdata/quilt/blob/master/catalog/app/utils/Data.js#L33